Assets/Project/Scripts/UnityChanControlScript.cs

```diff
@@ -54,178 +63,114 @@ private void Start()
 
         private void FixedUpdate()
         {
             SidewaysMovement();
-
-            // perform action depending on current state
-            currentBaseState = anim.GetCurrentAnimatorStateInfo(0);
-            ;
-            ActionsDependingOnAnimationState(currentBaseState.nameHash);
         }
 
-        #region Functions for Update()
-
-        private void SetAnimations()
-        {
-            horizontalInput = Input.GetAxis("Horizontal");
-            anim.SetFloat("Speed", verticalInput);
-            anim.SetFloat("Direction", horizontalInput);
-            anim.speed = animSpeed;
-        }
+        #region Functions for calling dictionary action depending on player input
 
-        private void JumpControls()
+        // detects all posible inputs we set and perform action if enabled
+        private void DetectInputs()
         {
-            rb.useGravity = true;
-
-            if (Input.GetButtonDown("Jump"))
+            if (Input.GetKeyDown(TryGetKeyCode(jumpParam)))
             {
-                rb.AddForce(Vector3.up * jumpPower, ForceMode.Impulse);
-                anim.SetBool("Jump", true);
+                PerformActionIfEnabled(jumpParam);
             }
-        }        
-        private void SlidingControls()
-        {
-            if (Input.GetKeyDown(KeyCode.S))
+            else if (Input.GetKeyDown(TryGetKeyCode(slideParam)))
             {
-                anim.SetBool("Slide", true);
+                PerformActionIfEnabled(slideParam);
             }
-        }
-        private void AttackControls()
-        {
-            if (Input.GetMouseButtonDown((int)MouseButtonDown.MBD_LEFT))
+            else if (Input.GetKeyDown(TryGetKeyCode(attackParam)))
             {
-                anim.SetBool("Attack", true);
+                PerformActionIfEnabled(attackParam);
             }
-        }
-        #endregion
-
-        #region Functions for FixedUpdate()
-
-        // enables player to only move move sideways
-        private void SidewaysMovement()
-        {
-            velocity = new Vector3(horizontalInput, 0, 0);
-            velocity = transform.TransformDirection(velocity);
-            velocity *= leftRightSpeed;
-            transform.localPosition += velocity * Time.fixedDeltaTime;
-        }
-
-        // performs action depending on current state of character
-        private void ActionsDependingOnAnimationState(int currentState)
-        {            
-            // check that dictionary has that state
-            if (animationStates.ContainsKey(currentState))
+            // tmp to test death animation
+            else if (Input.GetKeyDown(TryGetKeyCode(deathParam)))
             {
-                animationStates[currentState](); // invoke action in dictionary that corresponds to currentState
+                PerformActionIfEnabled(deathParam);
             }
-
-            return;
         }
 
-        void ResetCollider()
+        // return keycode if mapped in our dictionary
```

Comment:
Are all of these comments on functions necessary? I feel like most of these functions are pretty self documenting already 🎃